### ISSUE TITLE
Only one image can be attached to the 'Add new meal/drug' page

### ISSUE THREAD

**Issue by sdgun** [topcoderinc/va-kidney-android#141]
## Steps to Reproduce

1. Login as a valid user
2. Complete the profile data and click Generate Goals
3. Go to Home page
4. Tap Food/Drug > Add new Meal/drug
5. Attach a image
6. Now attach a new image (It will replace the previous image)

## Actual Result(s)

- Only one image can be attached to the 'Add new meal/drug' page

## Expected Result(s)

- User must able to attach multiple images to the 'Add new meal/drug' page, once user add 3-4 images to the Meal/drug and save the item,those images must display in the relevent item as 2x2 tiles
- Compare with the iOS funcionality

eg: If user attach 3 images, those 3 images must display as 2x2 tile in the Food Intake screen
eg: If user attach only one image, single image is displayed in the Food Intake screen
eg: If user attach 5 images, first 4 images must display as 2x2 tile in the Food Intake screen

## Frequency

- 5/5

## Environment Details

- **Device(s):** Samsung Galaxy S5
- **Operating System:** Android 6.0.1
- **Size:** 5.1 inches (~69.6% screen-to-body ratio)
- **Resolution:** 1080 x 1920 pixels (~432 ppi pixel density)
- **Browser(s):** Chrome 64.0.3282.137

## Image(s)/Video(s)/Console Log/Crash Log

![issue_20180914_120319](https://user-images.githubusercontent.com/4592712/45564930-9cd89b80-b817-11e8-9873-9155dfcb3bf2.png)

**Comment by nahidshahin** [comment]
@talesforce can I work on it?

**Comment by talesforce** [comment]
@nahidshahin - Sorry - no

We're doing this as a direct task with @KevinLeigh

**Comment by sdgun** [comment]
iOS:

![img_4556](https://user-images.githubusercontent.com/4592712/45581740-1e0c4e80-b869-11e8-85d2-13fdc9443c39.PNG)
![img_4555](https://user-images.githubusercontent.com/4592712/45581741-1e0c4e80-b869-11e8-9fdf-7dcf2195e005.PNG)

**Comment by KevinLeigh** [comment]
@sdgun Is there a limit to the amount of images a User can add ?

**Comment by sdgun** [comment]
@KevinLeigh I will check and confirm in few mins

**Comment by KevinLeigh** [comment]
@sdgun Thank you, please note I am using a linux distro currently and not Mac so I cannot easily run the IOS App in a simulator. Sorry about that.

**Comment by sdgun** [comment]
@KevinLeigh I think there is no limit, I have added 20. cc @talesforce @dustinTopcoder

**Comment by KevinLeigh** [comment]
Thank you for the response @sdgun.

**Comment by KevinLeigh** [comment]
@sdgun Hey there, should the App only completely remove meal/drug images when 'SAVE MEAL' button is clicked ?

**Comment by sdgun** [comment]
- Drug/Meals can be submitted/edited without having images @KevinLeigh

**Comment by sdgun** [comment]
@KevinLeigh Android works like below; 

If User presses back button, the images be permanently removed - **No**
or
User must click 'SAVE MEAL' button to permanently remove images - **Yes**

**Comment by KevinLeigh** [comment]
@sdgun Perfect, thank you. Will create a temporary removed collection to handle permentately removing images on save button click and not when back button is pressed. :)

**Comment by KevinLeigh** [comment]
@sdgun Please see PR here -> https://github.com/topcoderinc/va-kidney-android/pull/143

**Comment by KevinLeigh** [comment]
@sdgun Can a User replace an existing image with a new image by clicking on image and then selecting a new one from Gallery ?

**Comment by sdgun** [comment]
@KevinLeigh 

- When I create a new meal/drug without images, the placeholder must display in white color, Now I can see a default image there.
- When I add 3 images (Or deleting one by one and have 2 images) and tap on Save Meal will display the 3 images in the 2x2 tile window, if we edit that same item and remove 1 image from the list will display only one image in the 2x2 tile window. 
- Deleted images are still displayed again (Will provide a different video recording for this one, need to find exact steps here)

**Comment by sdgun** [comment]
@KevinLeigh Above Summary Updated

**Comment by KevinLeigh** [comment]
@sdgun So basically we should not display any placeholder image ? I basically just kept the existing placeholder image the App was using. You will need to click 'SAVE MEAL' each time if you want to finalize the adding or removing of an image, I will upload a video shortly to.

**Comment by KevinLeigh** [comment]
@sdgun For the video you uploaded -> https://cl.ly/b3258eb6184f, I do not see you adding any images by clicking add image.

**Comment by KevinLeigh** [comment]
@sdgun Please see demo video here -> https://drive.google.com/open?id=1yo08nwohjshhwxAFfdzwfxtpnOY1-4no

**Comment by sdgun** [comment]
So basically we should not display any placeholder image ? **Yes**

**Comment by sdgun** [comment]
@KevinLeigh I can see the same issue in your demo video (0:26-0:34) > You have added 2 images but in the 2x2 tile (main placeholder image) still displays only one.  Create a new meal/drug and add 3 images and save, check the 2x2 tile and it will display those 3 images on the meal/drug thumbnail

**Comment by KevinLeigh** [comment]
@sdgun The confusion good be that IOS is automatically attaching images based on the recommendations from API and that Android is not ? If that is the case the issue description above was not clear on that. I understand that you expected that I have a look at the IOS App, when developing multiple images feature, however I cannot run the IOS repo at the moment as I am using a linux distro.

**Comment by KevinLeigh** [comment]
@sdgun Yes I did, but issue description states ` once user add 3-4 images to the Meal/drug and save the item, those images must display in the relevent item as 2x2 tiles` So basically you will need to add up to 3 images to see 2x2 tiles. I presumed that we should only display 2x2 tiles when the User has added  3 or more images. If User adds less than 3 then we should only display one.

**Comment by sdgun** [comment]
Ohh, Sorry My bad

**Comment by sdgun** [comment]
- User add only one image > display the only image full size inside the meal/drug thumbnail
- User add two image > display two images on the meal/drug thumbnail (2x2) two images in the first row
- User add three image > display three images on the meal/drug thumbnail (2x2) two images in the first row + one image in the second row
- User add four or more image > display four images on the meal/drug thumbnail (2x2) two images in the first row + two image in the second row

Same when user deletes the images one by one

**Comment by KevinLeigh** [comment]
@sdgun All good, I have removed the placeholder image and now if a meal has no images attached then the App should display white background for Meal. 

Thank you for the updated description, I will display 2 x 2 images when there is more then one image attached :)

**Comment by KevinLeigh** [comment]
@sdgun Done and changes are pushed to repo. 2 x 2 tile images should always display if a User adds more then one image to a Meal

**Comment by sdgun** [comment]
@KevinLeigh FYI https://www.dropbox.com/s/nzesybxfuiokowr/TRMirage20180915190401.export.mp4?dl=0

**Comment by sdgun** [comment]
@KevinLeigh [1] ans [2] Fixed

- Deleted images are still displayed again 
01. Add 4 images 
02. Delete first 3 images
03. Add new 2 images (Now altogether there are 3 images)
04. Tap **Save meal** (When saving we have 3 images)
05. Check the meal/drug thumbnail
06. Tap on the item and go inside and observe (All the removed images are also there)

**Comment by KevinLeigh** [comment]
@sdgun I tried the above scenario and images delete fine. I did find out that if you add an image and remove image immediately after and the click 'SAVE MEAL', then the temporary added image is saved. I have fixed that issue and pushed up changes. I will record a video of the above scenario to show you that its working fine on my side.

**Comment by sdgun** [comment]
Fixed via https://github.com/topcoderinc/va-kidney-android/pull/143
